src/authorization/service/role.service.ts

```diff
@@ -30,10 +34,53 @@ export class RoleService {
     private permissionRepository: Repository<Permission>,
     private roleCacheService: RoleCacheService,
     private connection: Connection,
+    private searchService: SearchService,
   ) {}
 
-  async getAllRoles(): Promise<Role[]> {
-    return await this.rolesRepository.find();
+  async getAllRoles(input?: RoleInputFilter): Promise<Role[]> {
+    let searchTerm: { [key: string]: FindOperator<string | undefined> }[] = [];
```

Comment:
This logic is being rewritten across Users / Groups / Roles ... and any other future entities. Can we move this generic logic to SearchService?